docs/ql-style-guide.md

```diff
@@ -1,417 +1,425 @@
+1. Declarations that are deprecated *should* be documented as `DEPRECATED: ...`
+1. Declarations that are for internal use *should* be documented as `INTERNAL: Do not use`.
+
+### Examples
+
+```ql
+/** Provides logic for determining constant expressions. */
+```
+
+```ql
+/**
+ * Holds if the qualifier of this call has type `qualifierType`.
+ * `isExactType` indicates whether the type is exact, that is, whether
+ * the qualifier is guaranteed not to be a subtype of `qualifierType`.
+ */
+```
+```ql
+/**
+ * A delegate declaration, for example
+ * ```
+ * delegate void Logger(string text);
+ * ```
+ */
+class Delegate extends ...
+```
+
+```ql
+/**
+ * An element that can be called.
+ *
+ * Either a method (`Method`), a constructor (`Constructor`), a destructor
+ * (`Destructor`), an operator (`Operator`), an accessor (`Accessor`),
+ * an anonymous function (`AnonymousFunctionExpr`), or a local function
+ * (`LocalFunction`).
+ */
+class Callable extends ...
+```
+
+```ql
+/** DEPRECATED: Use `getAnExpr()` instead. */
+deprecated Expr getInitializer()
+```
+
+```ql
+/**
+ * INTERNAL: Do not use.
+ */
+```
+
+## Formulas
+1. *Prefer* one *conjunct* per line.
+1. Write the `and` at the end of the line. This also applies in `where` clauses.
+1. *Prefer* to write the `or` keyword on its own line.
+1. The `or` keyword *may* be written at the end of a line, or within a line, provided that it has no `and` operands.
+1. Single-line formulas *may* be used in order to save space or add clarity, particularly in the *body* of a *quantifier/aggregation*.
+1. *Always* use brackets to clarify the precedence of:
+   - `implies`
+   - `if`-`then`-`else`
+1. *Avoid* using brackets to clarify the precedence of:
+   - `not`
+   - `and`
+   - `or`
+1. Parenthesised formulas *can* be written:
+   - Within a single line. There *should not* be an additional space following the opening parenthesis or preceding the closing parenthesis.
+   - Spanning multiple lines. The opening parenthesis *should* be placed at the end of the preceding line, the body should be indented one level, and the closing bracket should be placed on a new line at the outer indentation.
+1. *Quantifiers/aggregations* *can* be written:
+   - Within a single line. In this case, there is no space to the inside of the parentheses, or after the quantifier keyword.
+   - Across multiple lines. In this case, type declarations are on the same line as the quantifier with the `|` at the end of the same line as the quantifier, the second `|` *must* be at the end of the same line as the quantifier or on its own line at the outer indentation, and the body of the quantifier *must* be indented one level. The closing `)` is written on a new line, at the outer indentation. If the type declarations need to be broken across multiple lines then there must *must* be a line break after the opening `(`, the type declarations indented one level, and the first `|` on its own line at the outer indentation.
```

Comment:
I'm not completely sure whether the `|` should be at the outer or the inner indentation, but agree that we need to specify it, and have the option of placing it on its own line.